### PR TITLE
[EOC-376] moving item between lists via socket

### PR DESCRIPTION
**Pull request by alemikolo**

### CODE REVIEW

File: server/controllers/list.js [hunk 1]
```diff
@@ -1646,13 +1645,18 @@ const moveItem = (req, resp) => {
       itemToUpdate.isDeleted = true;
       itemToUpdate.isArchived = true;
 
-      return returnPayload(oldList.save())({ newList, ...payload });
+      return returnPayload(oldList.save(), true)({ newList, newItem });
```
Comment (marekrozmus) [inline server/controllers/list.js]: proposals:
```
return oldList.save().then(res => ({ res, newList, newItem }));
return oldList.save().then(() => ({ newList, newItem }));

save().then(() => payload)

f = x => () => x
```

File: client/src/modules/list/components/Items/model/reducer.js [hunk 2]
```diff
@@ -145,16 +145,20 @@ const items = (state = {}, action) => {
         payload: { itemId, locks }
       } = action;
       const blockedItem = state[itemId];
-      const { locks: prevLocks } = blockedItem;
-      const updatedLocks = filterDefined(locks);
+      if (blockedItem) {
```
Comment (adamklepacz) [inline client/src/modules/list/components/Items/model/reducer.js]: line o a break above this line

File: server/sockets/helpers/index.js [hunk 3]
```diff
@@ -215,10 +215,33 @@ const emitRoleChange = (io, cohortClients, data, event) => {
   }
 };
 
+const emitListsMetaData = (io, cohortClients, dashboardClients, list) => {
```
Comment (adamklepacz) [inline server/sockets/helpers/index.js]: i would go for
```
const emitListsMetaData = io => (cohortClients, dashboardClients) => list  {....
```

File: server/sockets/helpers/index.js [hunk 4]
```diff
@@ -215,10 +215,33 @@ const emitRoleChange = (io, cohortClients, data, event) => {
...
+  const { _id: listId, cohortId, viewersIds } = list;
+
+  viewersIds.forEach(viewerId => {
+    const id = viewerId.toString();
+
+    if (dashboardClients.has(id)) {
+      const { socketId } = dashboardClients.get(id);
```
Comment (adamklepacz) [inline server/sockets/helpers/index.js]: line of a break

File: server/sockets/helpers/index.js [hunk 5]
```diff
@@ -215,10 +215,33 @@ const emitRoleChange = (io, cohortClients, data, event) => {
...
+      io.sockets.to(socketId).emit(ListActionTypes.FETCH_META_DATA_SUCCESS, {
+        [listId]: responseWithList(list, id)
+      });
+    }
+
+    if (cohortId && cohortClients.has(id)) {
+      const { socketId } = cohortClients.get(id);
```
Comment (adamklepacz) [inline server/sockets/helpers/index.js]: line of a break

File: server/sockets/list.js [hunk 6]
```diff
@@ -778,6 +779,44 @@ const restoreList = (
...
+
+  io.sockets.to(listChannel(listId)).emit(ItemActionTypes.DELETE_SUCCESS, {
+    listId,
+    itemId: oldItemId
+  });
+
+  viewersIds.forEach(viewerId => {
+    const id = viewerId.toString();
+
+    if (id !== userId.toString()) {
```
Comment (adamklepacz) [inline server/sockets/list.js]: why this if??

Comment (alemikolo) [reply server/sockets/list.js]: it was unnecessary - I removed it

File: server/sockets/helpers/index.js [hunk 7]
```diff
@@ -215,10 +215,35 @@ const emitRoleChange = (io, cohortClients, data, event) => {
   }
 };
 
+const emitListsMetaData = io => (cohortClients, dashboardClients) => list => {
```
Comment (adamklepacz) [inline server/sockets/helpers/index.js]: I can see that this method is duplicated. We already have.
Please check if you can merge it and use existing one

`updateListOnDashboardAndCohortView` in `server/sockets/helpers/index.js` file

File: server/sockets/list.js [hunk 8]
```diff
@@ -284,16 +262,13 @@ const clearVote = (io, listClients, viewersIds) => data => {
...
+    .to(listChannel(listId))
+    .emit(ListActionTypes.UPDATE_SUCCESS, { listId, ...rest });
 
-  updateListOnDashboardAndCohortView(
-    io.sockets,
-    listId,
-    dashboardViewClients,
-    cohortViewClients
-  );
+  emitListsMetaData(io)(cohortViewClients, dashboardViewClients)(list);
```
Comment (adamklepacz) [inline server/sockets/list.js]: I would go for `updateListOnDashboardAndCohortView`  name.
Application is getting bigger and this name is more descriptive.

**Comment by adamklepacz** [comment]
Remember to change the base to master before merging

**Pull request merged**